pkg/fleet/installer/exec/installer_exec.go

```diff
@@ -46,6 +46,8 @@ type installerCmd struct {
 
 func (i *InstallerExec) newInstallerCmd(ctx context.Context, command string, args ...string) *installerCmd {
 	env := i.env.ToEnv()
+	// Enforce the use of the installer when it is bundled with the agent.
+	env = append(env, "DD_BUNDLED_AGENT=installer")
```

Comment:
Can't we put that in `*env.Env` & set it in the agent-installer unit?

Reply:
I don't think it makes sense:
- This env variable is a detail to ensure the `installer` will be used if the binary we target is the core agent, regardless of symlink shenanigans. It makes sense to set it here, in the CLI wrapper.
- `env.Env` is really the "installer CLI config", not an umbrella of everything env var